src/main/java/net/fabricmc/loom/configuration/ifaceinject/InterfaceInjectionProcessor.java

```diff
@@ -210,10 +224,19 @@ public static List<InjectedInterface> fromMod(FabricModJson fabricModJson) {
 			final List<InjectedInterface> result = new ArrayList<>();
 
 			for (String className : addedIfaces.keySet()) {
-				final JsonArray ifaceNames = addedIfaces.getAsJsonArray(className);
+				final JsonArray ifacesInfo = addedIfaces.getAsJsonArray(className);
+
+				for (JsonElement ifaceElement : ifacesInfo) {
+					String ifaceInfo = ifaceElement.getAsString();
+
+					if (ifaceInfo.contains("<") && ifaceInfo.contains(">")) {
+						String ifaceName = ifaceInfo.substring(0, ifaceInfo.indexOf("<"));
+						String generics = ifaceInfo.substring(ifaceInfo.indexOf("<"));
 
-				for (JsonElement ifaceName : ifaceNames) {
-					result.add(new InjectedInterface(modId, className, ifaceName.getAsString()));
+						result.add(new InjectedInterface(modId, className, ifaceName, generics));
+					} else {
+						result.add(new InjectedInterface(modId, className, ifaceElement.getAsString(), null));
+					}
```

Comment:
Nit: drop the else, just have a single `new InjectedInterface` that passes a nullable generics String